SNAP-2917 - generating SparkR library along with snappy product

vatsalmevada opened the pull request:
## What changes were proposed in this pull request?
Removing some unused code here. Main changes available at https://github.com/SnappyDataInc/snappydata/pull/1286
## How was this patch tested?

Using R version `3.5.3`.
Manually launched `bin/sparkR` shell and tried some CRUD SQL queries using `sql()` function of sparkR.

## Other PRs
https://github.com/SnappyDataInc/snappydata/pull/1286

Commit:
SNAP-2917 - generating SparkR library along with snappy product

Commit:
adding dependency on generateRLibs task from sparkProduct task

vatsalmevada commented on assembly/build.gradle:
```diff
@@ -46,10 +46,19 @@ clean.doLast {
 
+task generateRLibs {
+  delete file("${sparkProjectRootDir}/R/lib")
+  if (!rootProject.hasProperty('R.skip')) {
```
should we also check whether the assigned value to this property is `true`? I noticed that we are not checking that for other properties like `snappydata.enterprise`.

ashetkar replied:
Either will do. I think that was done to avoid two checks (existence and value).

Commit:
Renaming the gradle property to R.enable from R.skip and changing the
behavior accordingly. By default R library won't be generated. In order
to generate the R library as part of snappy's spark distribution
R.enable property needs to be set as part of gradle.properties file.

Commit:
adding extra check to verify value of R.enable property

Commit:
Building SparkR using check-cran.sh script in order to match spark's
build process.
